### ISSUE TITLE
System.OutOfMemoryException: Optimization suggestions?

### ISSUE THREAD

**Issue by deivyd321** [SixLabors/ImageSharp#873]
Hello,
I have this method where I resize image(max size 10MB) to 1 MB.
 ```cs
public static Stream ResizeToTargetFileSize(IFormFile file, long targetBytes)
        {
            var imageFormat = ImageFormatFactory.GetImageFormat(file.ContentType, file.FileName);
            var outputStream = new MemoryStream();
            using (var stream = file.OpenReadStream())
            using (var image = Image.Load(stream))
            {
                if (stream.Length > targetBytes)
                {
                    var newSize = GetScaledSize(image, imageFormat.GetFactor(stream.Length, targetBytes));
                    image.Mutate(x => x.Resize(newSize));
                }
                image.Save(outputStream, imageFormat.Encoder);
            }
            outputStream.Position = 0;
            return outputStream;
        }
```

However, when I upload file 9.7MB size I am getting memory error. Are there any suggestions how to optimize memory by changing this code?
New Size is Width * 0.3, Height*0.3. 

The image: 
![Pizigani_1367_Chart_10MB](https://user-images.githubusercontent.com/17739850/55544166-2e83f980-56d3-11e9-887f-6a8ada21fbdf.jpg)

**Comment by JimBobSquarePants** [comment]
Please https://github.com/SixLabors/ImageSharp/issues/805

**Comment by antonfirsov** [comment]
also #642

**Comment by deivyd321** [comment]
I checked the mentioned issues but didn't find the proper solution or workaround. I am working on ASP .NET CORE app and when I upload and resize images(size 10 MB) the application memory instantly increases to 2-3GB would be there any suggestions how to avoid this? I have tried memory management methods from SixLabors/docs/blob/master/articles/ImageSharp/MemoryManagement.md page but it didn't seems to help. The code for resizing is the same as above.
![image](https://user-images.githubusercontent.com/17739850/55606279-defc0700-5780-11e9-92d8-313eb9b09138.png)
